bot/exts/moderation/silence.py

```diff
@@ -242,6 +242,11 @@ async def _set_silence_overwrites(self, channel: TextOrVoiceChannel, *, kick: bo
                 send_messages_in_threads=overwrite.send_messages_in_threads
             )
 
+        elif isinstance(channel, Thread):
+            log.info(f"Silencing thread #{channel.parent}/{channel} ({channel.parent.id}/{channel.id}).")
+            await channel.edit(locked=True)
```

Comment:
If the thread is already locked, this will just schedule an unlock in the set amount of time. Failing is probably better  to keep consistent with how it behaves with text channels.